xl-v1-utils.go

```diff
@@ -92,20 +89,18 @@ func readXLMeta(disk StorageAPI, bucket string, object string) ([]byte, error) {
 	var writer = new(bytes.Buffer)
 	startOffset := int64(0)
 
-	// Allocate 2MiB buffer, this is sufficient for the most of `xl.json`.
-	buf := make([]byte, 2*1024*1024)
-
 	// Read until io.EOF.
 	for {
```

Comment:
you can reuse readAll right?
`return readAll(disk, bucket, path.Join(object, xlMetaJSONFile))`